Pester returns errors even when no tests fail

### Where are you running it?
- Azure DevOps Service (VSTS)

### Version of Extension/Task
10.0.5

### Expected behaviour and actual behaviour
Expected: Pester task passes without error when no tests fail

Actual: Pester task fails with error even though all tests pass

### Steps to reproduce the problem
Here's a link to a public project where this issue can be repro'd

https://dev.azure.com/cfarrelly/test/_build/results?buildId=4&view=logs&j=275f1d19-1bd8-5591-b06b-07d489ea915a&t=f969aa47-2434-555b-2817-6587663ed32a

I thought originally it may be down to tags and skipping tests but I created a simple repro here with 1 test to run and the test runner fails with 'Pester returned errors', appears to be an issue when checking the results here

https://github.com/pester/AzureDevOpsExtension/blob/e2141f28ee869c892dfc91e3191145da14f125d3/Extension/PesterTask/PesterV10/Pester.ps1#L145

Thanks,

Colin.

Just hit this bug as well...

```
2020-06-03T12:15:39.5996504Z Tests Passed: 3, 
2020-06-03T12:15:39.6033071Z Failed: 0, 
2020-06-03T12:15:39.6070344Z Skipped: 0 
2020-06-03T12:15:39.6150264Z NotRun: 0
2020-06-03T12:15:39.6180214Z 
2020-06-03T12:15:39.6181136Z 
2020-06-03T12:15:40.2186133Z ##[error]C:\Build\CI-Agent2\_work\_tasks\Pester_cca5462b-887d-4617-bf3f-dcf0d3c622e9\10.0.5\Pester.ps1 : Pester returned errors
At line:1 char:1

2020-06-03T12:15:40.2198914Z ##[error]C:\Build\CI-Agent2\_work\_tasks\Pester_cca5462b-887d-4617-bf3f-dcf0d3c622e9\10.0.5\Pester.ps1 : Pester returned errors
At line:1 char:1

2020-06-03T12:15:40.2217307Z ##[error]+ C:\Build\CI-Agent2\_work\_tasks\Pester_cca5462b-887d-4617-bf3f-dcf0d3 ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Pester.ps1
 

2020-06-03T12:15:40.2220096Z ##[error]+ C:\Build\CI-Agent2\_work\_tasks\Pester_cca5462b-887d-4617-bf3f-dcf0d3 ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Pester.ps1
 

2020-06-03T12:15:40.2716171Z Pester Script finished
2020-06-03T12:15:40.2875525Z ##[section]Finishing: Pester Test Runner
```

I too faced the same error and waiting for the solution. Please find below for the raw error log report.

**Also the test result is showing as passed even though I intentionally placed the code to fail the test.**

2020-06-04T10:22:55.0696725Z ##[error]D:\a\_tasks\Pester_cca5462b-887d-4617-bf3f-dcf0d3c622e9\10.0.5\Pester.ps1 : Pester returned errors
At line:1 char:1
+ d:\a\_tasks\Pester_cca5462b-887d-4617-bf3f-dcf0d3c622e9\10.0.5\Pester ...

2020-06-04T10:22:55.0709967Z ##[error]D:\a\_tasks\Pester_cca5462b-887d-4617-bf3f-dcf0d3c622e9\10.0.5\Pester.ps1 : Pester returned errors
At line:1 char:1
+ d:\a\_tasks\Pester_cca5462b-887d-4617-bf3f-dcf0d3c622e9\10.0.5\Pester ...

2020-06-04T10:22:55.0729293Z ##[error]+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Pester.ps1
 

2020-06-04T10:22:55.0731667Z ##[error]+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Pester.ps1
 

2020-06-04T10:22:55.1061073Z Pester Script finished
2020-06-04T10:22:55.1140399Z ##[section]Finishing: Pester Test Runner

--- a/Extension/PesterTask/PesterV10/Pester.ps1
+++ b/Extension/PesterTask/PesterV10/Pester.ps1
@@ -92,6 +92,7 @@ $PesterConfig = @{
     Run = @{
         Path = $TestFolder
         Exit = $true
+        PassThrue = $true
     }
     TestResult = @{
         Enabled = $true
@@ -142,6 +143,6 @@ if (-not([String]::IsNullOrWhiteSpace($ScriptBlock))) {
 
 $result = Invoke-Pester -Configuration  ([PesterConfiguration]$PesterConfig)
 
-if ($result.failedCount -ne 0) {
+if ($result.failedCount -gt 0) {
     Write-Error "Pester returned errors"
 }

--- a/Extension/PesterTask/PesterV10/Pester.ps1
+++ b/Extension/PesterTask/PesterV10/Pester.ps1
@@ -91,8 +91,7 @@ $PesterConfig = @{
 
     Run = @{
         Path = $TestFolder
-        Exit = $true
-        PassThrue = $true
+        PassThru = $true
     }
     TestResult = @{
         Enabled = $true
@@ -143,6 +142,6 @@ if (-not([String]::IsNullOrWhiteSpace($ScriptBlock))) {
 
 $result = Invoke-Pester -Configuration  ([PesterConfiguration]$PesterConfig)
 
-if ($result.failedCount -gt 0) {
-    Write-Error "Pester returned errors"
+if ($Result.FailedCount -gt 0) {
+    Write-Error "Pester Failed at least one test. Please see results for details."
 }

--- a/pester-v10.yml
+++ b/pester-v10.yml
@@ -17,6 +17,7 @@ steps:
 
   - task: PublishTestResults@2
     displayName: 'Publish Test Results Test-Pester.XML'
+    condition: 'always()'
     inputs:
       testRunTitle: '$(agent.os)-v10'
       testResultsFormat: NUnit
@@ -24,6 +25,7 @@ steps:
     
   - task: PublishCodeCoverageResults@1
     displayName: 'Publish code coverage'
+    condition: 'always()'
     inputs:
       summaryFileLocation: '$(System.DefaultWorkingDirectory)/CC-Pester.XML'
   

--- a/pester-v9.yml
+++ b/pester-v9.yml
@@ -17,6 +17,7 @@ steps:
 
   - task: PublishTestResults@2
     displayName: 'Publish Test Results Test-Pester.XML'
+    condition: 'always()'
     inputs:
       testRunTitle: '$(agent.os)-v9'
       testResultsFormat: NUnit
@@ -24,6 +25,7 @@ steps:
     
   - task: PublishCodeCoverageResults@1
     displayName: 'Publish code coverage'
+    condition: 'always()'
     inputs:
       summaryFileLocation: '$(System.DefaultWorkingDirectory)/CC-Pester.XML'
   

--- a/testResults.xml
+++ b/testResults.xml
@@ -1,0 +1,29 @@
+﻿<?xml version="1.0" encoding="utf-8" standalone="no"?>
+<test-results xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="nunit_schema_2.5.xsd" name="Pester" total="6" errors="0" failures="1" not-run="0" inconclusive="0" ignored="0" skipped="0" invalid="0" date="2020-06-07" time="17:19:09">
+  <environment platform="Microsoft Windows 10 Enterprise|C:\WINDOWS|\Device\Harddisk0\Partition4" os-version="10.0.19041" nunit-version="2.5.8.0" machine-name="DESKTOP-T1JUALL" cwd="C:\source\github\PesterAzureDevOpsExtension" user-domain="DESKTOP-T1JUALL" clr-version="Unknown" user="Chris" />
+  <culture-info current-culture="en-GB" current-uiculture="en-GB" />
+  <test-suite type="TestFixture" name="Pester" executed="True" result="Failure" success="False" time="0.1278" asserts="0" description="Pester">
+    <results>
+      <test-suite type="TestFixture" name="C:\source\github\PesterAzureDevOpsExtension\Extension\tests\Calculator\Add-Numbers.Tests.ps1" executed="True" result="Failure" success="False" time="0.1278" asserts="0" description="C:\source\github\PesterAzureDevOpsExtension\Extension\tests\Calculator\Add-Numbers.Tests.ps1">
+        <results>
+          <test-suite type="TestFixture" name="Add-Numbers" executed="True" result="Failure" success="False" time="0.1126" asserts="0" description="Add-Numbers">
+            <results>
+              <test-case description="adds positive numbers" name="Add-Numbers.adds positive numbers" time="0.0027" asserts="0" success="True" result="Success" executed="True" />
+              <test-case description="adds negative numbers" name="Add-Numbers.adds negative numbers" time="0.0057" asserts="0" success="True" result="Success" executed="True" />
+              <test-case description="adds one negative number to positive number" name="Add-Numbers.adds one negative number to positive number" time="0.0029" asserts="0" success="True" result="Success" executed="True" />
+              <test-case description="concatenates strings if given strings" name="Add-Numbers.concatenates strings if given strings" time="0.0085" asserts="0" success="True" result="Success" executed="True" />
+              <test-case description="should not be 0" name="Add-Numbers.should not be 0" time="0.003" asserts="0" success="True" result="Success" executed="True" />
+              <test-case description="Should fail" name="Add-Numbers.Should fail" time="0.0235" asserts="0" success="False" result="Failure" executed="True">
+                <failure>
+                  <message>Expected $false, but got $true.
+at $true | Should -Be $false, C:\source\github\PesterAzureDevOpsExtension\Extension\tests\Calculator\Add-Numbers.Tests.ps1:28</message>
+                  <stack-trace>at &lt;ScriptBlock&gt;, C:\source\github\PesterAzureDevOpsExtension\Extension\tests\Calculator\Add-Numbers.Tests.ps1:28</stack-trace>
+                </failure>
+              </test-case>
+            </results>
+          </test-suite>
+        </results>
+      </test-suite>
+    </results>
+  </test-suite>
+</test-results>

